core/src/main/java/io/kestra/core/services/FlowService.java

```diff
@@ -39,6 +43,45 @@ public Stream<Flow> keepLastVersion(Stream<Flow> stream) {
         return keepLastVersionCollector(stream);
     }
 
+    public List<String> deprecationPaths(Flow flow) {
+        List<String> deprecationPaths = deprecationTraversal("", flow).toList();
+        return deprecationPaths.isEmpty() ? null : deprecationPaths;
+    }
+
+    private Stream<String> deprecationTraversal(String prefix, Object object) {
+        if(ClassUtils.isPrimitiveOrWrapper(object.getClass()) || String.class.equals(object.getClass())) {
+            return Stream.empty();
+        }
+
+        return Stream.concat(
```

Comment:
I'm not very comfortable to use reflection here, the JsonSchema of the flow already have deprecation information could we use this instead?